### PR TITLE
React: Dropdown for lab selection

### PR DESCRIPTION
**Pull request by Lakshya2610**
- Added a dropdown component to allow for lab selection
- On lab selection, the custom endpoint is saved in `APIInfo`
- Default value for endpoint in `APIInfo` is null which implies the app should use the default endpoint

Note: Still need to do some refactoring to make use of this custom endpoint value

#908

### CODE REVIEW

**Review by cklamann — CHANGES_REQUESTED** [review]
I can see the lab selection control on the login screen, though I don't see any options and the login text controls are hidden. I think that if we want to merge this in as a WIP, it would be best to hide the dropdown, probably just conditioned on the number of endpoint options available (which right now would be 0).

**Comment by Lakshya2610** [comment]
I added the default "Select a lab" option at the very end which broke the conditional rendering. Fixed it now

File: react/src/App.tsx [hunk 1]
```diff
@@ -85,11 +85,20 @@ function BaseApp(props: { darkMode: boolean; toggleDarkMode: () => void }) {
 
             const apiInfoResult = await fetch("/api");
             if (apiInfoResult.ok) {
-                const apiInfo = await apiInfoResult.json();
+                let apiInfo = await apiInfoResult.json();
+                if (!("endpoint" in apiInfo)) {
+                    apiInfo = { ...apiInfo, endpoint: null }; // null endpoint value -> use the default endpoint
+                }
```
Comment (kevinlul) [inline react/src/App.tsx:91]: Are you supposing that the correct endpoint (domain or prefix) for this tenant will be in the response from the `/api` endpoint on the current domain?

Comment (Lakshya2610) [reply react/src/App.tsx:91]: This was an assumption initially that @cklamann cleared up. That statement doesn't need to be wrapped in an if condition.

Comment (kevinlul) [reply react/src/App.tsx:91]: No, the problem is that you can't make a request to the backend to figure out what backend to make requests to!

Comment (Lakshya2610) [reply react/src/App.tsx:91]: Yeah, this code was initially done with the incorrect assumption. I'm gonna remove this for now as it'll be fixed in the second PR for this feature. This PR is for purely UI stuff

**Comment by kevinlul** [comment]
The result after adding items to `getLabs()`
image

**Pull request closed without merging**